### PR TITLE
add documentation on input checks for hlf contracts

### PR DESCRIPTION
**Pull request by nklsd**
*** Reason for this PR ***
For better understanding on valid inputs, checks, which are performed on inputs, need to be documented.

*** Changes in this PR ***
Added listing of the method parameters which are checked as well as a short description of the corresponding checks in the according markdown document of each contract.

### CODE REVIEW

File: hlf/chaincode/matriculation.md [hunk 1]
```diff
@@ -27,7 +27,8 @@ The Errors returned are defined [here](errors.md#Errors).
         ]
       }
       ```
-       - Description: This error is returned, if the given parameters could not be parsed. If some attributes within the json are not well formatted, they are listed in invalidParams.
+       - Description: This error is returned, if the given parameters could not be parsed. If some attributes within the json are not well formatted, they are listed in invalidParams.  
+       For detailed informations see [matriculationData checks](#matriculationDataCecks).
```
Comment (nklsd) [inline hlf/chaincode/matriculation.md]:
```suggestion
       For detailed informations see [matriculationData checks](#matriculationDataChecks).

```

**Review by NikoBergemann — APPROVED** [review]

File: hlf/chaincode/matriculation.md [hunk 2]
```diff
@@ -130,6 +134,7 @@ This method adds a single entry to the list of semesters in the MatriculationDat
       }
       ```
       - Description: This error is returned, if the given parameters could not be parsed. If some attributes within the json are not well formatted, they are listed in invalidParams.
+      For detailed informations see [subjectMatriculation checks](#subjectMatriculationCecks).
```
Comment (nklsd) [inline hlf/chaincode/matriculation.md]:
```suggestion
      For detailed informations see [subjectMatriculation checks](#subjectMatriculationChecks).

```

File: hlf/chaincode/matriculation.md [hunk 3]
```diff
@@ -69,7 +72,8 @@ The Errors returned are defined [here](errors.md#Errors).
         ]
       }
       ```
-      - Description: This error is returned, if the given parameters could not be parsed. If some attributes within the json are not well formatted, they are listed in invalidParams.
+      - Description: This error is returned, if the given parameters could not be parsed. If some attributes within the json are not well formatted, they are listed in invalidParams.  
+      For detailed informations see [matriculationData checks](#matriculationDataCecks).
```
Comment (nklsd) [inline hlf/chaincode/matriculation.md]:
```suggestion
      For detailed informations see [matriculationData checks](#matriculationDataChecks).

```

**Review by matthias-geuchen — APPROVED** [review]

File: hlf/chaincode/matriculation.md [hunk 4]
```diff
@@ -166,3 +174,22 @@ This method adds a single entry to the list of semesters in the MatriculationDat
...
+
+
+### <a id="subjectMatriculationChecks" />subjectMatriculation
+- Check, if not null or an empty list.
+- **fieldOfStudy**
+  - Check, if not null or an empty String.
+  - Check for duplicate entries.
+- **semesters**
+  - Check, if not null or an empty list.
+  - Check, if containing strings are in the format (WS\d{4}/\d{2}|SS\d{4})
```
Comment (bastihav) [inline hlf/chaincode/matriculation.md]: Do you only check the format? Is WS2020/23 valid for you?

**Review by bastihav — APPROVED** [review]

File: hlf/chaincode/matriculation.md (hunk 4 above)
Comment (nklsd) [reply hlf/chaincode/matriculation.md]: No, but apparently we forgot to document it. We also check, if the years contained in the winter semester are consecutive and in the right order. I added the docu on that.

**Pull request merged**